MWSE/LuaObject.h

```diff
@@ -576,6 +577,81 @@ namespace mwse::lua {
 		}
 	};
 
+	template<>
+	class ObjectCreator<TES3::Book> : public ObjectCreatorBase {
+	public:
+		TES3::BaseObject* create(sol::table params, bool getIfExists) const override {
+			std::string id = getOptionalParam<std::string>(params, "id", {});
+
+			if (id.size() > 31)
+				throw std::invalid_argument{ "tes3.createObject: 'id' parameter must be less than 32 character long." };
+
+			auto existingObject = TES3::DataHandler::get()->nonDynamicData->resolveObject(id.c_str());
+			if (existingObject) {
+				if (existingObject->objectType == TES3::ObjectType::Book) {
+					return existingObject;
+				}
+				else {
+					throw std::invalid_argument{ "tes3.createObject: 'id' parameter already assigned to an existing object that is not a book." };
+				}
+			}
+
+			std::string name = getOptionalParam<std::string>(params, "name", "Book");
+			if (name.size() > 31)
+				throw std::invalid_argument{ "tes3.createObject: 'name' parameter must be less than 32 character long." };
+
+			std::string mesh = getOptionalParam<std::string>(params, "mesh", {});
+			if (mesh.size() > 31)
+				throw std::invalid_argument{ "tes3.createObject: 'mesh' parameter must be less than 32 character long." };
+
+			auto book = new TES3::Book();
+
+			book->setID(id.c_str());
+			book->setName(name.c_str());
+			book->setModelPath(mesh.c_str());
+
+			auto script = getOptionalParamScript(params, "script");
+
+			if (script != nullptr)
+				book->script = script;
+
+			auto enchantment = getOptionalParamObject<TES3::Enchantment>(params, "enchantment");
+			auto enchantCapacity = getOptionalParam<unsigned short>(params, "enchantCapacity", 0);
+			auto bookType = getOptionalParam<unsigned char>(params, "type", 0);
+			if (enchantment && enchantment->objectType == TES3::ObjectType::Enchantment) {
+				book->enchantment = enchantment;
+			}
+			else if (enchantCapacity > 0 && bookType == 1)
```

Comment:
You probably want to create an enum for this one in TES3Book.h. You could see `TES3::SpellEffectState` for an example.

Reply:
Done. Let me know if there's any issues with it.

Reply:
Looks fine to me, thanks!